src/runtime-pmix.c

```diff
@@ -132,6 +159,47 @@ shmem_runtime_exchange(void)
     bool wantit=true;
     //bool active = true;
 
+    if (node_ranks) {
+        pmix_proc_t proc;
+        pmix_value_t *val;
+
+        PMIX_LOAD_NSPACE(proc.nspace, myproc.nspace);
+        proc.rank = PMIX_RANK_WILDCARD;
+
+        if (PMIX_SUCCESS == (rc = PMIx_Get(&proc, PMIX_LOCAL_SIZE, NULL, 0, &val))) {
+            node_size = val->data.uint32;
+            PMIX_VALUE_RELEASE(val);
+            if (node_size <= 0)
+                RETURN_ERROR_MSG_PREINIT("Invalid PMIX_LOCAL_SIZE (%d)\n", node_size);
+        } else {
+            RETURN_ERROR_MSG_PREINIT("PMIX_LOCAL_SIZE is not properly initiated (%d)\n", rc);
+        }
+
+        node_ranks = (int *)malloc(size * sizeof(int));
+
+        for (int i = 0; i < size; i++) {
+            node_ranks[i] = -1;
+        }
+
+        /* Note: PMIX_LOCAL_PROCS should be available in the near future (would avoid parsing) */
```

Comment:
Looks like `PMIX_LOCAL_PROCS` might be ready for use.  Should we create an issue for this to address later?

Reply:
See #846